src/cpp/r/session/RSessionState.cpp

```diff
@@ -224,6 +225,20 @@ Error restoreEnvironmentVars(const FilePath& envFile)
 
    return Success();
 }
+
+Error executeAfterRestartCommand(const FilePath& afterRestartFile)
+{
+   std::string command;
+   Error error = core::readStringFromFile(afterRestartFile, &command);
+   if (error)
+      LOG_ERROR(error);
+   
+   if (command.empty())
+      return Success();
+   
+   Rprintf("> %s\n", command.c_str());
```

Comment:
We might want a better way of simulating the prompt here, or just use a separate prefix (`>>>`) to indicate this is a restart command.